tracekit.js

```diff
@@ -149,27 +160,31 @@ TraceKit.report = (function reportModuleWrapper() {
      * @memberof TraceKit.report
      * @throws An exception if an error occurs while calling an handler.
      */
-    function notifyHandlers(stack, isWindowError, error) {
+    function notifyHandlers(stack, isWindowError, aArguments) {
         var exception = null;
         if (isWindowError && !TraceKit.collectWindowErrors) {
           return;
         }
         for (var i in handlers) {
-            if (_has(handlers, i)) {
+        if (_has(handlers, i) && handlers[i][1] === TraceKit.windowPointer) {
                 try {
-                    handlers[i](stack, isWindowError, error);
+                  var errorObj=(aArguments.length > 4) ? aArguments[4] : null;
+                  handlers[i][0](stack, isWindowError, errorObj);
                 } catch (inner) {
                     exception = inner;
                 }
+          // Call old onerror events
+          if (handlers[i][2]) {
```

Comment:
We weren't doing this before were we? It's just really hard to follow this and make sense of what's going on..  We call all the other handlers in a try catch.